diff --git a/cocos2dx/cocoa/CCArray.cpp b/cocos2dx/cocoa/CCArray.cpp
index 12e5bb2971..0a916cd6e2 100644
--- a/cocos2dx/cocoa/CCArray.cpp
+++ b/cocos2dx/cocoa/CCArray.cpp
@@ -134,7 +134,7 @@ CCArray* CCArray::createWithContentsOfFile(const char* pFileName)
 
 CCArray* CCArray::createWithContentsOfFileThreadSafe(const char* pFileName)
 {
-    return CCFileUtils::sharedFileUtils()->createCCArrayWithContentsOfFile(pFileName);
+    return nullptr;
 }
 
 bool CCArray::init()
diff --git a/cocos2dx/cocoa/CCInteger.h b/cocos2dx/cocoa/CCInteger.h
index f700a30a35..20afae5a92 100644
--- a/cocos2dx/cocoa/CCInteger.h
+++ b/cocos2dx/cocoa/CCInteger.h
@@ -30,7 +30,6 @@ public:
      */
     virtual void acceptVisitor(CCDataVisitor &visitor) { visitor.visit(this); }
 
-private:
     int m_nValue;
 };
 
diff --git a/cocos2dx/cocoa/CCObject.cpp b/cocos2dx/cocoa/CCObject.cpp
index dd74c695a3..a2b5f8b574 100644
--- a/cocos2dx/cocoa/CCObject.cpp
+++ b/cocos2dx/cocoa/CCObject.cpp
@@ -59,15 +59,9 @@ CCObject::~CCObject(void)
     // if the object is referenced by Lua engine, remove it
     if (m_nLuaID)
     {
-        CCScriptEngineManager::sharedManager()->getScriptEngine()->removeScriptObjectByCCObject(this);
     }
     else
     {
-        CCScriptEngineProtocol* pEngine = CCScriptEngineManager::sharedManager()->getScriptEngine();
-        if (pEngine != NULL && pEngine->getScriptType() == kScriptTypeJavascript)
-        {
-            pEngine->removeScriptObjectByCCObject(this);
-        }
     }
 }
 
diff --git a/cocos2dx/support/CCNotificationCenter.cpp b/cocos2dx/support/CCNotificationCenter.cpp
index 835854eded..b3b75fc59d 100644
--- a/cocos2dx/support/CCNotificationCenter.cpp
+++ b/cocos2dx/support/CCNotificationCenter.cpp
@@ -177,8 +177,6 @@ void CCNotificationCenter::postNotification(const char *name, CCObject *object)
         {
             if (0 != observer->getHandler())
             {
-                CCScriptEngineProtocol* engine = CCScriptEngineManager::sharedManager()->getScriptEngine();
-                engine->executeNotificationEvent(this, name);
             }
             else
             {
